### PR TITLE
ci(release): track deprecations in release notes

### PR DESCRIPTION
**Pull request by w-bonelli**
* add script `doc/mf6io/mf6ivar/deprecations.py` to scrape DFNs and generate markdown table
* run deprecations script in `docs.yml` CI and upload result file as artifact
* add `mk_deprecations.py` script to convert markdown to LaTeX table
* update `build_docs.py` to include deprecations in release notes
* add deprecation policy section to `DEVELOPER.md`
* add deprecations page to ReadTheDocs

### CODE REVIEW

**Comment by w-bonelli** [comment]
Inserts a new section Deprecations into release notes, e.g.

Screenshot 2023-08-08 at 12 33 11 PM

sample from test run [here](https://github.com/w-bonelli/modflow6-nightly-build/actions/runs/5798749309)

File: DEVELOPER.md [hunk 1]
```diff
@@ -576,3 +576,19 @@ git branch -d feat-xyz-backup
...
+
+To deprecate a MODFLOW 6 input/output option in a DFN file:
+
+- Add a line `deprecated x.y.z` to the option block in the appropriate DFN file, where `x.y.z` is the version the deprecation is introduced. Mention the deprecation prominently in the release notes.
+- If support for the deprecated option is removed (typically after at least 2 minor or major releases or 1 year), add a line `removed x.y.z` to the options block, where `x.y.z` is the version in which support for the option was removed. Mention the removal prominently in the release notes.
+- Deprecated/removed options are not removed from DFN files but remain in perpetuity. The `doc/mf6io/mf6ivar/mf6ivar.py` script generates a markdown deprecation table which is converted to LaTeX by `doc/ReleaseNotes/mk_deprecations.py` for inclusion in the MODFLOW 6 release notes. Deprecations and removals should still be mentioned separately in the release notes, however.
+
+### Finding deprecations
+
+To search for `deprecated` (or substitute `removed`) options with `git`:
```
Comment (langevin-usgs) [inline DEVELOPER.md:590]:
```suggestion
To search for `deprecated` (or substitute `removed`) strings in DFN files with `git`:
```

**Review by langevin-usgs — COMMENTED** [review]
Nice job.  This looks ready to me.  Minor edits included in this review.

**Comment by w-bonelli** [comment]
thanks @langevin-usgs edits applied.

**Review by langevin-usgs — APPROVED** [review]

File: doc/mf6io/mf6ivar/md/deprecations.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#### Deprecations
+
+The following table lists deprecated options and the versions in which they were deprecated and (optionally) removed.
+
+| Model-Package | Option | Deprecated | Removed |
+|:--------------|:-------|:-----------|:--------|
+| gwf-mvr | unit_conversion | 6.4.2 |  |
```
Comment (jdhughes-usgs) [inline doc/mf6io/mf6ivar/md/deprecations.md:7]: None of these options are actually in the `gwf-mvr` package

**Pull request merged**